web/ui/src/Instances/instancesFactory.js

```diff
@@ -3,106 +3,46 @@
 (function() {
     'use strict';
 
-    var instanceMap = {},
-        instanceList = [],
-        // make angular share with everybody!
-        resourcesFactory, $q, serviceHealth;
-
-    var UPDATE_FREQUENCY = 3000,
-        updatePromise;
+    var resourcesFactory, $q, serviceHealth;
 
     angular.module('instancesFactory', []).
-    factory("instancesFactory", ["$rootScope", "$q", "resourcesFactory", "$interval", "$serviceHealth",
-    function($rootScope, q, _resourcesFactory, $interval, _serviceHealth){
+    factory("instancesFactory", ["$rootScope", "$q", "resourcesFactory", "$interval", "$serviceHealth", "baseFactory",
+    function($rootScope, q, _resourcesFactory, $interval, _serviceHealth, BaseFactory){
 
         // share resourcesFactory throughout
         resourcesFactory = _resourcesFactory;
         $q = q;
         serviceHealth = _serviceHealth;
 
-        // public interface for instancesFactory
-        // TODO - evaluate what should be exposed
-        return {
-            // returns an instance by id
-            get: function(id){
-                return instanceMap[id];
-            },
+        var newFactory = new BaseFactory(Instance, resourcesFactory.get_running_services);
+
+        // alias some stuff for ease of use
+        newFactory.instanceArr = newFactory.objArr;
+        newFactory.instanceMap = newFactory.objMap;
 
-            getByServiceId: (id) => {
+        angular.extend(newFactory, {
+            getByServiceId: function(id){
                 let results = [];
-                for(let i in instanceMap){
-                    if(instanceMap[i].model.ServiceID === id){
-                        results.push(instanceMap[i]);
+                for(let i in this.objMap){
```

Comment:
Why replace `instanceMap` with `objMap`? 

Within a given file, I think it would be better to use the same naming convention.  Also, if I look for `instanceMap`, the only reference I find is line 21, which seems a little crufty